tests/cpp/include/test_util.h

```diff
@@ -70,11 +70,9 @@ class BlobMemory {
     return handle_.dptr;
   }
   void Free() {
-    if (handle_.dptr) {
-      Storage *storage = mxnet::Storage::Get();
-      storage->DirectFree(handle_);
-      handle_.dptr = nullptr;
-    }
+    mxnet::Storage::Get()->DirectFree(handle_);
+    handle_.size = 0;
+    handle_.dptr = nullptr;
```

Comment:
can you add a test for allocating size 0 blob and checking if handle_.dptr == nullptr. Otherwise LGTM

Reply:
This is a very nice suggestion!

Reply:
Good point. Will add an unit test.